### PR TITLE
Bmc

### PR DESCRIPTION
**Pull request by logicminds**
this adds support the the foreman proxy code to access the smart proxy bmc feature

### CODE REVIEW

File: lib/proxy_api.rb [hunk 1]
```diff
@@ -1,18 +1,21 @@
 require "rest_client"
 require "json"
 require "uri"
-require 'net'
```
Comment (ohadlevy) [inline lib/proxy_api.rb:4]: hmm.. why?

File: lib/proxy_api.rb [hunk 2]
```diff
@@ -1,18 +1,21 @@
...
+require 'net/http'
+require 'net/https'
```
Comment (ohadlevy) [inline lib/proxy_api.rb]: are you using Net:: directly? why not rest_client?

File: lib/proxy_api.rb (hunk 1 above)
Comment (logicminds) [inline lib/proxy_api.rb:4]: The compiler complained without it.  Said it couldn't find net but found
net/http.

Corey

Comment (ohadlevy) [inline lib/proxy_api.rb:4]: interesting, net is located at /lib/net.rb :)

File: lib/proxy_api.rb (hunk 2 above)
Comment (logicminds) [inline lib/proxy_api.rb]: It appears that net is never used.  I have removed the requirement.

File: lib/proxy_api.rb (hunk 1 above)
Comment (logicminds) [inline lib/proxy_api.rb:4]: turns out net is not even needed.  So I have removed it completely.

File: lib/proxy_api.rb [hunk 3]
```diff
@@ -1,18 +1,20 @@
...
   class Resource
     attr_reader :url
+    attr_accessor :connect_params
 
     def initialize(args)
       raise("Must provide a protocol and host when initialising a smart-proxy connection") unless (url =~ /^http/)
 
       # Each request is limited to 60 seconds
-      connect_params = {:timeout => 60, :open_timeout => 10, :headers => { :accept => :json }}
+      @connect_params = {:timeout => 60, :open_timeout => 10, :headers => { :accept => :json },
```
Comment (ohadlevy) [inline lib/proxy_api.rb]: if its attr_accessor, you dont need to @

**Comment by ohadlevy** [comment]
merged with minor changes - thanks @logicminds!

**Pull request closed without merging**